src/plugins/es_ui_shared/server/license/license.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import {
+  Logger,
+  KibanaRequest,
+  KibanaResponseFactory,
+  RequestHandler,
+  RequestHandlerContext,
+} from 'src/core/server';
+
+/* eslint-disable @kbn/eslint/no-restricted-paths */
+import type {
+  ILicense,
+  LicenseType,
+  LicenseCheckState,
+} from '../../../../../x-pack/plugins/licensing/common/types';
+import type { LicensingPluginStart } from '../../../../../x-pack/plugins/licensing/server/types';
+/* eslint-enable @kbn/eslint/no-restricted-paths */
+
+type LicenseLogger = Pick<Logger, 'warn'>;
+type LicenseDependency = Pick<LicensingPluginStart, 'license$'>;
+
+interface SetupSettings {
+  pluginName: string;
+  logger: LicenseLogger;
+}
+
+interface StartSettings {
+  pluginId: string;
+  minimumLicenseType: LicenseType;
+  licensing: LicenseDependency;
+}
+
+export class License {
+  private pluginName?: string;
+  private logger?: LicenseLogger;
+  private licenseCheckState: LicenseCheckState = 'unavailable';
+  private licenseType?: LicenseType;
+
+  private _isEsSecurityEnabled: boolean = false;
+
+  setup({ pluginName, logger }: SetupSettings) {
+    this.pluginName = pluginName;
+    this.logger = logger;
+  }
+
+  start({ pluginId, minimumLicenseType, licensing }: StartSettings) {
+    licensing.license$.subscribe((license: ILicense) => {
+      this.licenseType = license.type;
+      this.licenseCheckState = license.check(pluginId, minimumLicenseType!).state;
+
+      // Retrieving security checks the results of GET /_xpack as well as license state,
+      // so we're also checking whether security is disabled in elasticsearch.yml.
+      this._isEsSecurityEnabled = license.getFeature('security').isEnabled;
+    });
+  }
+
+  getLicenseErrorMessage(licenseCheckState: LicenseCheckState): string {
```

Comment:
nit; if this does not need to be public I would mark it as `private`.

Reply:
Good idea!